src/rpc/blockchain.cpp

```diff
@@ -604,8 +607,11 @@ static std::vector<uint8_t> GetRawBlockChecked(BlockManager& blockman, const CBl
     FlatFilePos pos{};
     {
         LOCK(cs_main);
-        if (blockman.IsBlockPruned(blockindex)) {
-            throw JSONRPCError(RPC_MISC_ERROR, "Block not available (pruned data)");
+        if (!(blockindex.nStatus & BLOCK_HAVE_DATA)) {
+            if (blockman.IsBlockPruned(blockindex)) {
+                throw JSONRPCError(RPC_MISC_ERROR, "Block not available (pruned data)");
+            }
+            throw JSONRPCError(RPC_MISC_ERROR, "Block not available (not fully downloaded)");
```

Comment:
nit: Same for here

        // Block not found on disk. This could be because we have the block
        // header in our index but not yet have the block or did not accept the
        // block. Or if the block was pruned right after we released the lock above.

Reply:
done, thanks!